bootstrap/frontend/src/main/scala/io/udash/bootstrap/datepicker/UdashDatePicker.scala

```diff
@@ -164,32 +165,44 @@ class UdashDatePicker private[datepicker](val date: Property[ju.Date],
     )
   }
 
-  private def dateToMomentString(date: ju.Date): String =
-    Try(moment(options.get.locale.getOrElse("en"), date.getTime, "x").format(options.get.format)).getOrElse(null)
+  private def internalFormat = options.get.format
+  private def internalLocale = options.get.locale.getOrElse("en")
+
+  private def dateToMomentString(date: Option[ju.Date]): String =
+    date.map(d => moment(internalLocale, d.getTime, "x").format(internalFormat)).getOrElse("")
 
   private def dateToMoment(date: ju.Date): MomentFormatWrapper =
-    Try(moment(options.get.locale.getOrElse("en"), date.getTime, "x")).getOrElse(null)
+    Try(moment("en", date.getTime, "x")).getOrElse(null)
 
-  private def momentStringToDate(date: String): ju.Date =
-    Try(new ju.Date(moment(options.get.locale.getOrElse("en"), date, options.get.format).valueOf().toLong)).getOrElse(null)
+  private def momentStringToDate(date: String): Option[ju.Date] =
+    moment(internalLocale, date, internalFormat).valueOf() match {
+      case t if t.isNaN || t.isInfinity => None
+      case t => Some(new ju.Date(t.toLong))
+    }
 
   private def momentToDate(date: MomentFormatWrapper): ju.Date =
```

Comment:
This one shares the validation logic with `momentStringToDate`. Keep it DRY :sun_with_face: